### ISSUE TITLE
Crash with RAR5 archive

### ISSUE THREAD

**Issue by RowlaxX** [borisbrodski/sevenzipjbinding#26]
Hello, I have a java program that download RAR5 files and then extract them with your library. It's working correctly for hundreds of files, exept for this file. This file contains two items, and only the first one is extractable, the second one, when extracting, cause the JVM crashing.

When I use an external program like 7Zip, the extraction works good.

Here is the link of the file :
https://d4cjuq.am.files.1drv.com/y4mJMhky-MQnlTmVlLD8Nu8jtpjM24blPAEgBuASPdp_S0soTqM44pdu9ZJMwLHO5ih81bhjI3nqVr1qhMsbuMwfMTi1BrrUtzMUvXy1rwpao9RJeTnyVVeWH5ViXTewhJAbA5iL2Of3M-uQi7MY5jwv4dkf8i3wDB6TTNdzJfEziYlVY-M1isxB3ZCiMT-JMwb/UnRenamed%20Files%20-%20Commodore%208Bits%20(2020-04-25).rar?download&psid=1

[hs_err_pid10868.log](https://github.com/borisbrodski/sevenzipjbinding/files/4590024/hs_err_pid10868.log)

**Comment by borisbrodski** [comment]
Hello RowlaxX,

thank you for reporting the issue.
I will look into it and come back to you.

Cheers,
Boris Brodski

**Comment by RowlaxX** [comment]
Thank you !
I will upload the file to another host if you are not able to grab it

**Comment by borisbrodski** [comment]
Yes please, upload it to some other cloud, so I can download it. Currently I get 404.

**Comment by RowlaxX** [comment]
Here you have
https://www.dropbox.com/s/6mw7zs8hkozsnly/UnRenamed%20Files%20-%20Commodore%208Bits%20%282020-04-25%29.rar?dl=0

**Comment by jinfeihan57** [comment]
As I know the p7zip 16.02 has some CVE bug about RAR archive. but I fix
CVE-2018-5996 <https://nvd.nist.gov/vuln/detail/CVE-2018-5996> maybe it
will fix your problem.you can try  https://github.com/szcnick/p7zip . （ I
can‘t download it.  and test it.but maybe you can test it . ）Test if your
problem is solved.

RowlaxX <notifications@github.com> 于2020年5月11日周一 下午4:39写道：

**Comment by borisbrodski** [comment]
I can confirm, that your file crashes p7zip 16.02, so it seems to be the 7-zip problem, not the binding problem.

**Comment by jinfeihan57** [comment]
@borisbrodski 
can you try this p7zip 17.02. https://github.com/szcnick/p7zip

**Comment by borisbrodski** [comment]
p7zip 17.02 can extract the archive 👍

I will upgrade 7-Zip-JBinding to the 17.02 version.
But it has also to pass all a couple thousands of my tests though. ;)

**Comment by RowlaxX** [comment]
I donated you a few bucks. This is not a huge ammount, but that’s a beer ! 😉Thank you for your hard work ! Provenance : Courrier pour Windows 10 De : Boris BrodskiEnvoyé le :mardi 1 septembre 2020 10:30À : borisbrodski/sevenzipjbindingCc : RowlaxX; AuthorObjet :Re: [borisbrodski/sevenzipjbinding] Crash with RAR5 archive (#26) p7zip 17.02 can extract the archive 👍I will upgrade 7-Zip-JBinding to the 17.02 version.But it has also to pass all a couple thousands of my tests though. ;)—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by borisbrodski** [comment]
Thank you very much! I highly appreciate it 🥇

**Comment by bartektartanus** [comment]
I'm also getting an error while trying to extract RAR5 archive:
```
FATAL ERROR: _getInstanceFromObject(): 'jobject' can't be null
Crash jvm to get a stack trace
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x0000000127b713cd, pid=66147, tid=0x0000000000000f03
#
# JRE version: Java(TM) SE Runtime Environment (8.0_131-b11) (build 1.8.0_131-b11)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.131-b11 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [lib7-Zip-JBinding.dylib+0x43cd]  fatal(char const*, ...)+0xcd
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# hs_err_pid66147.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```

@borisbrodski so when can we expect new version with updated p7zip? :)

**Comment by sunny-shu** [comment]
@RowlaxX 
the download url is broken, i cannot download the file.
@borisbrodski 
i am a member of p7zip，i will test this project with new version.

**Comment by borisbrodski** [comment]
@bartektartanus Thank you for posting it here :)

Can you please check, if you can or can not extract your RAR file with p7zip 16.02?
It p7zip 16.02 crashes, could you try to extract your RAR with p7zip 17.02 from here (https://github.com/szcnick/p7zip)
or upload your RAR file, so I can check it.

(This crash looks a bit weird to me. Maybe there is some other problem.)

Thank you very much!

**Comment by sunny-shu** [comment]
@bartektartanus 
p7zip 17.01 is based on p7zip 16.02 and fix the CVES，It is compatible with 16.02。
you can replace the source code with p7zip 17.01, then build the project 'sevenzipjbinding' and test if it is ok.

p7zip 17.02 update the 7z version to 17.01,files and apis changes a lot;
so it will fail when building sevenzipjbinding with p7zip 17.02.

**Comment by borisbrodski** [comment]
@sunny-shu good to know. Thank you. I will integrate p7zip and 7zip 17.02 in a while, when I'm done expanding my test infrastructure.

**Comment by borisbrodski** [comment]
**23.01-2.2** upgrades the engine to **7-Zip 23.01** (much newer RAR5 handling) and fixes several native issues. Could you re-test that file with 23.01-2.2 and let us know whether the JVM crash on the second item still happens? It's on Maven Central, [GitHub Releases](https://github.com/borisbrodski/sevenzipjbinding/releases/tag/Release-23.01-2.2) and SourceForge. Leaving open pending your check.
